int8 num
float32 percents